lib/sles4sap/azure_cli.pm

```diff
@@ -1384,14 +1391,53 @@ sub az_resource_delete {
     croak "Arguments 'name' and 'ids' are mutually exclusive" if $args{ids} and $args{name};
     croak "Argument 'name' or 'ids' has to be specified" unless $args{ids} or $args{name};
     $args{timeout} //= 60;
+    $args{temp_shell_script} //= '/tmp/az_resource_delete.sh';
 
     my @az_command = ('az resource delete',
         "--resource-group $args{resource_group}"
     );
     push @az_command, "--name $args{name}" if $args{name};
     push @az_command, "--ids $args{ids}" if $args{ids};
+    push @az_command, "--verbose" if $args{verbose};
+
+    # az cli command has to be run in a shell script. Problem with `script_run` api call is that it does not handle
+    # long commands very well. If the command is too long, internal `wait_serial` call might fail.
+    write_sut_file($args{temp_shell_script}, join(' ', @az_command));
+    assert_script_run("chmod +x $args{temp_shell_script}");
+    return script_run("/usr/bin/bash $args{temp_shell_script}", timeout => $args{timeout});
+}
+
+=head2 az_resource_list
+
+    az_resource_list([resource_group=>$resource_group, query=>$query, output=>$output]);
+
+Lists existing az resources based on arguments provided. Calling function without any argument returns full information
+from all existing resource groups.
+Returns decoded json structure if json format is requested, otherwise whole output is a string.
+
+=over 3
+
+=item B<resource_group> Existing resource group name.
+
+=item B<query> Jmespath query
+
+=item B<output> Output format. Default: 'json'
```

Comment:
I propose to remove output argument and just code the function like

```
sub az_resource_list {
    my (%args) = @_;

    my @az_command = ('az resource list');
    push(@az_command, "--resource-group $args{resource_group}") if $args{resource_group};
    push(@az_command, "--query \"$args{query}\"") if $args{query};
    push(@az_command, "--output json");

    my $result = script_output(join(' ', @az_command));
    return (decode_json($result));
}
```

Reply:
I counter-propose to keep it, but extend it with:
```
use YAML::PP;
...
    return (decode_json($result) if($args{output} eq 'json');
    if ($args{output} eq 'yaml') {
        my $yp = YAML::PP->new(schema => [qw/ + Merge /]);
        return $yp->load_string($result);
    }
    return ($result);
...

Reply:
I removed it but @mpagot promised he will implement it if needed :P